### DISCUSSION TITLE
How does one change the timeout of a service in a test?

### DISCUSSION THREAD

**Discussion by saul-jb** [feathersjs/feathers#2543] [Q&A] [answered]
How does one change the timeout of a service in a test?

The does not work in the test despite working on the client:
```javascript
const app = require("../src/app");

app.service("myService").timeout = 15000;
```

### Steps to reproduce

1. Setup a feathers app.
2. Generate a custom service and in one of its methods delay the response for longer than 5000ms.
3. Setup a test to test this method inside the generated app.test.js.
4. Add the line `app.service("myService").timeout = 15000;` at the top.
5. Get the error on the test `Timeout: Timeout of 5000ms exceeded calling create on myService`

### Expected behavior
The test should pass.

### Actual behavior
The test fails because of `Timeout: Timeout of 5000ms exceeded calling create on myService`.

### System configuration

Tell us about the applicable parts of your setup.

**Feathers version**:
4.7.0
**NodeJS version**:
v16.13.1

**Comment by daffl** [comment]
This is the timeout of the testing framework you are using. With Mocha and a normal setup you would add e.g. `--timeout 30000` in https://github.com/feathersjs/feathers-chat/blob/crow/package.json#L33.

**Reply by saul-jb** [reply]
This does not appear to be the case, I am using jest and have given a generous timeout but still have the error referring to a timeout of only 5000ms.

**Reply by daffl** [reply]
Are you using the client in your tests as well then? How is it being set up?

**Reply by saul-jb** [reply]
Oh that's it, I was attempting to change the timeout on the server app object instead of the client.

```javascript
const app = feathers();

// Timeout needs to be set on the client object.
app.service("myService").timeout = 15000;
```

Thanks for your help.

**Comment by saul-jb** [comment, marked as answer]
Timeout needs to be set on the client object not the server object:

```javascript
const feathers = require("@feathersjs/feathers");
const app = require("../src/app");

const client = feathers();

// Timeout needs to be set on the client object.
client.service("myService").timeout = 15000;

// This will not work
app.service("myService").timeout = 15000;
```
